Add more information about failure reasons (#119)
vippsas/vipps-recurring-api: vipps-recurring-api.md

| 5 | `REFUNDED` | The charge has been refunded. The timeframe for issuing a refund is 365 days from the date of capture. |
| 6 | `PARTIALLY_REFUNDED`| A part of the captured amount has been refunded. |
| 7 | `RESERVED` | An initial charge with `transactionType` set to `RESERVE_CAPTURE` changes state to `CHARGED` when captured successfully. |
| 8 | `CANCELLED` | The charge has been cancelled. |
| 9 | `PROCESSING` | The charge is currently being processed by Vipps. |

### Charge failure reasons

When fetching a charge through the API, you can find two fields in the response body to identify why the charge failed `failureReason` and `failureDescription`:

```json
{
  "status": "FAILED",
  "type": "RECURRING",
  "failureReason": "insufficient_funds",
  "failureDescription": "Payment was declined by the payer bank due to lack of funds"
}
```

Here is a list of possible values for `failureReason`, their respective descriptions and possible actions that the user/merchant could take.

| Reason | Description | Action |
| ---- | ----------- | ------ |

> | insufficient_funds | Payment was declined by the payer bank due to lack of funds. | User should fund the card to cover the difference between the amount to be paid or use another payment source that is adequately funded to complete the transaction. |

Reviewer: ```suggestion
| insufficient_funds | Payment was declined by the payer bank due to lack of funds. | User must either add funds to the card to cover the difference between the amount to be paid. Alternatively they can change to another, or add a new, payment source that is adequately funded to complete the transaction. |
```

> | invalid_card | The user tried to pay using a card that has either expired or is disabled by the issuer. | User must change payment source |

Reviewer: ```suggestion
| invalid_card | The user tried to pay using a card that has either expired or is disabled by the issuer. | User must change, or add a new, payment source on the agreement in Vipps. |
```

> | verification_required | Payment declined because the issuing bank requires verification. | |

Reviewer: ```suggestion
| verification_required | Payment declined because the issuing bank requires verification. | User should try to change, or add a new, payment source on their agreement in Vipps. Alternatively deleting their current card and adding it again might solve the issue. |
```

> | verification_required | Payment declined because the issuing bank requires verification. | |

Reviewer: ```suggestion
| verification_required | Payment declined because the issuing bank requires verification. | Ask the user to change, or add a new, payment source on their agreement in Vipps. Alternatively removing and then adding the card might solve the issue. |
```

| invalid_payment_source | The provided payment source is disabled or does not exist. | User must change payment source for the agreement. | User must change payment source

> | internal_error | Internal Error / Something went wrong | The error couldn not be identified as one of the above. Contact Vipps for more information. |

Reviewer: ```suggestion
| internal_error | Internal Error / Something went wrong | The error could not be identified as one of the above. Try to create the charge again, changing or adding payment sources on the agreement, or contact Vipps for more information. |
```
